flow/connectors/mongo/mongo.go

```diff
@@ -405,3 +409,41 @@ func defaultPipeline() mongo.Pipeline {
 		}}},
 	}
 }
+
+func parseAsClientOptions(config *protos.MongoConfig) (*options.ClientOptions, error) {
+	connStr, err := connstring.Parse(config.Uri)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing uri: %w", err)
+	}
+
+	clientOptions := options.Client().
+		ApplyURI(config.Uri).
+		SetAppName("PeerDB Mongo Connector").
+		SetAuth(options.Credential{
+			Username: config.Username,
+			Password: config.Password,
+		}).
+		// always use compression
+		SetCompressors([]string{"zstd", "snappy"}).
+		// always use majority read concern for correctness
+		SetReadConcern(readconcern.Majority())
+
+	// allow user to override with other read preference
+	if connStr.ReadPreference == "" {
+		clientOptions.SetReadPreference(readpref.SecondaryPreferred())
+	}
+
+	if !config.DisableTls && len(connStr.Hosts) > 0 {
+		tlsConfig, err := shared.CreateTlsConfig(tls.VersionTLS12, config.RootCa, connStr.Hosts[0], config.TlsHost, false)
```

Comment:
I'm not sure about using `.Hosts[0]` here if provided a list of hosts. What does the connector do here? Can tls not work if we connect to a different host?